Fix build failure on PowerPc architecture

giuliobenetti opened the pull request:
In PowerPc Linux only get_ra() exists[0] and conflicts with local get_ra() that has a completely different purpose. So let's rename local get_ra() to wifi_get_ra() to make it different from Linux's get_ra().

[0]: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/arch/powerpc/include/asm

Signed-off-by: Giulio Benetti <giulio.benetti@benettiengineering.com>

lucaceresoli reviewed (changes requested):
Hi Giulio,

the patch seems good, thanks.

I would change this in the commmit message though:

"In PowerPc Linux only get_ra() exists" -> "get_ra() exists only on PowerPC"

And the link you provide points to the powerpc directory, which is a bit generic. It would be mor informative if it pointed to the file and line where get_ra is defined, I think: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/arch/powerpc/include/asm/disassemble.h?h=v6.1-rc7#n49

CGarces commented on rtw_odm.c:
```diff
@@ -283,7 +283,7 @@ void rtw_odm_parse_rx_phy_status_chinfo(union recv_frame *rframe, u8 *phys)
 					, *phys & 0xf
 					, MAC_ARG(get_ta(wlanhdr))
-					, is_broadcast_mac_addr(get_ra(wlanhdr)) ? "BC" : is_multicast_mac_addr(get_ra(wlanhdr)) ? "MC" : "UC"
+					, is_broadcast_mac_addr(wifi_get_ra(wlanhdr)) ? "BC" : is_multicast_mac_addr(get_ra(wlanhdr)) ? "MC" : "UC"
```
Why the call from is_multicast_mac_addr is not renamed?

giuliobenetti replied:
Good catch, I don't know how it could compile correctly...
Now I fix it and push --force.

Thank you!

giuliobenetti commented:
> Hi Giulio,
> 
> the patch seems good, thanks.
> 
> I would change this in the commmit message though:
> 
> "In PowerPc Linux only get_ra() exists" -> "get_ra() exists only on PowerPC"
> 
> And the link you provide points to the powerpc directory, which is a bit generic. It would be mor informative if it pointed to the file and line where get_ra is defined, I think: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/arch/powerpc/include/asm/disassemble.h?h=v6.1-rc7#n49



> Hi Giulio,
> 
> the patch seems good, thanks.
> 
> I would change this in the commmit message though:
> 
> "In PowerPc Linux only get_ra() exists" -> "get_ra() exists only on PowerPC"

Ok

> And the link you provide points to the powerpc directory, which is a bit generic. It would be mor informative if it pointed to the file and line where get_ra is defined, I think: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/arch/powerpc/include/asm/disassemble.h?h=v6.1-rc7#n49

Oops, when I've copied and paste it's been truncated.

Thank you!

lwfinger commented:
I merged this manually. Do a git pull.

giuliobenetti commented:
@lwfinger I've checked but it's not complete. Diff your patch:
https://github.com/lwfinger/rtl8188eu/commit/f0cc43f5b250c13f97610d3e343ae557512bcc1b
with mine:
https://github.com/lwfinger/rtl8188eu/pull/432/commits/129685e7c775b8ce5cd0ed65bdbb04a9fb85e6f6
You've forgotten some file. Also my authorship :-)

lwfinger commented:
commit f0cc43f5b250c13f97610d3e343ae557512bcc1b (HEAD -> v5.2.2.4, origin/v5.2.2.4)
Author: Larry Finger <Larry.Finger@lwfinger.net>
Date:   Mon Nov 28 13:36:55 2022 -0600

    rtl8188eu: Change get_ra() to wifi_get_ra()
    This is the commit message. I did not forget your contribution

    In PowerPc Linux only get_ra() exists[0] and conflicts with local get_ra()
    that has a completely different purpose. So let's rename local get_ra()
    to wifi_get_ra() to make it different from Linux's get_ra().
    
    [0]: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/arch/powerpc/include/asm/disassemble.h?h=v6.1-rc7#n49
    
    Signed-off-by: Giulio Benetti <giulio.benetti@benettiengineering.com>
    Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>

giuliobenetti commented:
> commit [f0cc43f](https://github.com/lwfinger/rtl8188eu/commit/f0cc43f5b250c13f97610d3e343ae557512bcc1b) (HEAD -> v5.2.2.4, origin/v5.2.2.4) Author: Larry Finger [Larry.Finger@lwfinger.net](mailto:Larry.Finger@lwfinger.net) Date: Mon Nov 28 13:36:55 2022 -0600
> 
> ```
> rtl8188eu: Change get_ra() to wifi_get_ra()
> This is the commit message. I did not forget your contribution
> 
> In PowerPc Linux only get_ra() exists[0] and conflicts with local get_ra()
> that has a completely different purpose. So let's rename local get_ra()
> to wifi_get_ra() to make it different from Linux's get_ra().
> 
> [0]: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/arch/powerpc/include/asm/disassemble.h?h=v6.1-rc7#n49
> 
> Signed-off-by: Giulio Benetti <giulio.benetti@benettiengineering.com>
> Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
> ```

Don’t mind. Anyway some code is missing.

giuliobenetti commented:
> commit [f0cc43f](https://github.com/lwfinger/rtl8188eu/commit/f0cc43f5b250c13f97610d3e343ae557512bcc1b) (HEAD -> v5.2.2.4, origin/v5.2.2.4) Author: Larry Finger [Larry.Finger@lwfinger.net](mailto:Larry.Finger@lwfinger.net) Date: Mon Nov 28 13:36:55 2022 -0600
> 
> ```
> rtl8188eu: Change get_ra() to wifi_get_ra()
> This is the commit message. I did not forget your contribution
> 
> In PowerPc Linux only get_ra() exists[0] and conflicts with local get_ra()
> that has a completely different purpose. So let's rename local get_ra()
> to wifi_get_ra() to make it different from Linux's get_ra().
> 
> [0]: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/arch/powerpc/include/asm/disassemble.h?h=v6.1-rc7#n49
> 
> Signed-off-by: Giulio Benetti <giulio.benetti@benettiengineering.com>
> Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
> ```

FYI if you want to keep the original authorship you need to amend the commit like this:
$ git commit —amend —author=“Giulio Benetti <giulio.benetti@benettiengineering.com>”
That way the authorship will change.
Signed-off-by is another thing, not the authorship.
But again, no problem :-)

lwfinger commented:
I got the other file. In the kernel, the leading s-o-b is the author. Your suggest, but I added you as author. Your git command did not work. The first thing is that "amend" needs "--amend". I do not know what formatting the Author line needs. I added it to the commit and pushed it.

giuliobenetti commented:
> I got the other file. In the kernel, the leading s-o-b is the author. Your suggest, but I added you as author. Your git command did not work. The first thing is that "amend" needs "--amend". I do not know what formatting the Author line needs. I added it to the commit and pushed it.

Again, it's not a problem for me, but what you write is not correct @lwfinger: the first SoB is not the Author for git and so neither in Linux. SoB and Author is not the same thing, they are separate.
Try to do this test on one of your repository:
```
$ echo hello > hello.txt
$ git add hello.txt
$ git commit -s -m "Hello commit"
$ git log
```
The output of "git log" will give you(with your "Name Surname <e-mail>":
```
commit 33577d684fd7abae71ef20d682c70f500d25a82c (HEAD -> master)
Author: Giulio Benetti <giulio.benetti@benettiengineering.com>
Date:   Mon Nov 28 23:11:42 2022 +0100

    Hello commit
    
    Signed-off-by: Giulio Benetti <giulio.benetti@benettiengineering.com>
```
But as you can see there is the field "Author" AND the SoB and they are not the same thing.
So, if you can't simply commit another user patch, otherwise it will be committed with your authorship(note Author) with the other SoB.
To avoid this you need to first commit the patch and then to rebase edit that commit and then you need to issue the command above.
Above I've forgotten to add ``` for pointing you the shell command as well as the angular parenthesis to my e-mail. The correct shell command is:
```
$ git commit —amend —author=“Giulio Benetti <giulio.benetti@benettiengineering.com>"
```
So, to complete, let's say you want to give Authorship to "John Doe <john.doe@doe.com>", you have to:
```
$ git rebase -i HEAD~1
replace 'pick' with 'edit' and quit-save
$ git commit --amend --author="John Doe <john.doe@doe.com>"
```
This way you will end up having the Authorship of John Doe and my SoB, then to fix that you have to rebase reword that commit and you're done.

I hope everything is clear now and you appreciate the explanation!
